pytest_fixtures/satellite_auth.py

```diff
@@ -265,31 +267,53 @@ def enable_external_auth_rhsso(enroll_configure_rhsso_external_auth):
 def enroll_idm_and_configure_external_auth():
     """Enroll the Satellite6 Server to an IDM Server."""
     run_command(cmd='yum -y --disableplugin=foreman-protector install ipa-client ipa-admintools')
-
     run_command(
         cmd=f'echo {settings.ipa.password} | kinit admin',
         hostname=settings.ipa.hostname,
     )
-    result = run_command(
-        cmd=f'ipa host-add --random {settings.server.hostname}',
-        hostname=settings.ipa.hostname,
-    )
+    output = ssh.command(f'ipa host-find {settings.server.hostname}')
+    if output.return_code != 0:
+        result = run_command(
+            cmd=f'ipa host-add --random {settings.server.hostname}',
```

Comment:
Please don't reference `settings.server.hostname` in fixtures or tests directly, unless absolutely necessary.

There is a `default_sat` fixture, which will provide you a satellite instance. It has a `hostname` property that you can reference here, ensuring that when this fixture runs its executing ssh commands against the correct target satellite.

Reply:
@pondrejk as discussed, I'm not going to enforce this change.  But note that you very well may encounter BoxKeyErrors on these tests while this fixture is referencing `settings.server.hostname` directly. The resolution for this is in using `default_sat`, which @JacobCallahan or I will apply in the 'unstable' branch.